specs/lazy-adr/adr-011-based-sequencing.md

```diff
@@ -0,0 +1,188 @@
+
+### State Preconf and Forkchoice rule
+
+Proposer is not necessary for based sequencing rollups, however they can be used for efficiency purposes. Every node in the rollup network will follow a global forkchoice rule for forming batches (aka computing rollup state updates) using the selected & ordered (by the base layer) transactions. A simple forkchoice rule can simply be based on the max bytes or gas limits of the execution layer. For example, every batch needs to be under 2 megabytes in size or 2 million gas units. Gas estimations can be complicated at the sequencing layer of Rollkit, hence we use max bytes, where every continuous batch does not exceed max bytes in size.
+
+While every rollkit node can independently prepare the batch by itself (meaning every node is a proposer) and if all of them follow the global forkchoice rule, the state updates should be consistent across all the nodes. An alternative implementation could simple elect a proposer (similar to single sequencer mode) with state preconf to quickly accept the p2p shared batches (by the proposer) along with state updates and inclusion proof data to later hard confirm. Note that, this optimization differs from the single sequencer in key ways listed below:
+- the full nodes prepare batches according to the global forkchoice rule in the background and not solely rely on the proposer. This will help catch if the proposer skips any transaction in the base layer.
+- the full nodes can also relay the user transaction to base layer unlike in case of single sequencer where the proposer batches user transactions which then later published to the data availability layer.
+- no trust requirement on the proposer as every full node does the full verification (which includes, preparing the batches from the base layer by following the global forkchoice rule, applies the batch to validate the state update, validates the proposer shared batch data which includes the inclusions proofs of the base layer transactions used to prepare the batch).
+
+### Rollup Blocks (or Batches) from Base layer blobs
+
+A rollup block (or batch, aka header) can be prepared using one or more base layer blobs that contains rollup transactions. At the same time, multiple rollup blocks can be formed using a base layer blob that contains rollup transactions. This is purely dependent on the global forkchoice rule that is selected for forming the batches. In our case it is the max bytes that the execution layer can support, hence, a batch , up to the max bytes may be formed using the rollup transactions in the base layer.
```

Comment:
[question]: is there a height associated to this? how do execution apis know which height is for which txs?